NOnion/CircuitState.fs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NOnion
+
+open System.Threading.Tasks
+
+open NOnion.Crypto
+
+type CircuitState =
+    | Initialized
+    | CreatingFast of
+        circuitId: uint16 *
+        randomClientMaterial: array<byte> *
+        completionTask: TaskCompletionSource<uint16>
+    | Created of circuitId: uint16 * cryptoState: TorCryptoState
+    //TODO:
```

Comment:
pls elaborate comment, to do what?

Reply:
The items below the todo are not handled yet, I will explain it better